Forge/src/main/java/me/ferdz/placeableitems/client/AllVertexBoundingBox.java

```diff
@@ -0,0 +1,267 @@
+package me.ferdz.placeableitems.client;
+
+import net.minecraft.client.renderer.Vector3d;
+import net.minecraft.util.math.AxisAlignedBB;
+
+/**
+ * Represents a box defined by 8 points in the world. Unlike {@link AxisAlignedBB}s, points
+ * are not validated. Therefore, caution must be taken when creating instances of this box
+ * as some points may not be properly named according to this class' member methods. The
+ * methods are named according to their position in the world relative to the player when facing
+ * the bounding box.
+ *
+ * @author Parker Hawke - Choco
+ */
+public class AllVertexBoundingBox {
```

Comment:
I'm confused as to why you added this. Wouldn't it be better to use AxisAlignedBB or anything that's already built in? There must be something built-in that supports rotating voxels, right?

Reply:
Because the fluid model can be rotated and the AxisAlignedBB only holds 2 opposing points of a cuboid, it's impossible to properly calculate the position of the other 6 unidentified points. You would only be able to rotate the 2 points in the AABB which would result in the two corners being rotated, but the other 6 points being relative to those 2 points.

For example:
- A bounding box at rotation zero with bounds (0, 0, 0) -> (1, 1, 1) would have the back, right left point of (1, 0, 1). This is correct.
- The same bounding box rotated 45 degrees would have the points (0, 0, 0.5) -> (1, 1, 1.5) would have the back, right left point of (1, 0, 1.5) which is incorrect and result in the box rendering as a flat line between the two points.

Therefore, all 8 points must be defined and rotated accordingly. I was previously using an AABB and ran into this issue, so I know for certain this is the only solution. I could not find an existing class in the vanilla or Forge codebase though I'd be happily proven wrong.

Reply:
Okay, I looked quickly and didn't find an alternative either